### PR TITLE
Add annex business logic

### PR DESCRIPTION
**Pull request by meaboutsoftware**
This PR contains changes related to the action of adding annex to binding contract:

- Annex can be only added when binding contract is neither expired not terminated
- Annex is added from binding contract context
- There is no expiration date on annex. When binding contract expires, then annex is as well expired

Additionally, it contains some cleanups in existing entities, inconsistent namings and removal of redundant qualifiers.

Application, infra and API logic will be added in the next PRs.

### CODE REVIEW

File: Chapter-4-applying-tactical-domain-driven-design/Fitnet.Contracts/Src/Fitnet.Contracts.Core/AddAnnex/AnnexAddedToBindingContractEvent.cs [hunk 1]
```diff
@@ -0,0 +1,22 @@
+﻿namespace EvolutionaryArchitecture.Fitnet.Contracts.Core.AddAnex;
```
Comment (kamilbaczek) [inline Chapter-4-applying-tactical-domain-driven-design/Fitnet.Contracts/Src/Fitnet.Contracts.Core/AddAnnex/AnnexAddedToBindingContractEvent.cs:1]:
```csharp
namespace EvolutionaryArchitecture.Fitnet.Contracts.Core.AddAnnex;
```

Please fix in all new files

File: Chapter-4-applying-tactical-domain-driven-design/Fitnet.Contracts/Src/Fitnet.Contracts.Core/AddAnnex/BusinessRules/AnnexCanOnlyBeAddedOnlyBeAddedToActiveBindingContractRule.cs [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    {
+        _terminatedAt = terminatedAt;
+        _expiringAt = expiringAt;
+    }
+
+    public bool IsMet() => !_terminatedAt.HasValue && _expiringAt > DateTimeOffset.UtcNow;
+
+    public string Error => "Annex can only be added to active binding contract";
+}
+
```
Comment (kamilbaczek) [inline Chapter-4-applying-tactical-domain-driven-design/Fitnet.Contracts/Src/Fitnet.Contracts.Core/AddAnnex/BusinessRules/AnnexCanOnlyBeAddedOnlyBeAddedToActiveBindingContractRule.cs:22]: Please remove unessery enter

File: Chapter-4-applying-tactical-domain-driven-design/Fitnet.Contracts/Src/Fitnet.Contracts.Core/AddAnnex/BusinessRules/AnnexCanOnlyBeAddedOnlyBeAddedToActiveBindingContractRule.cs [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+    internal AnnexCanOnlyBeAddedOnlyBeAddedToActiveBindingContractRule(
+        DateTimeOffset? terminatedAt,
+        DateTimeOffset expiringAt)
+    {
+        _terminatedAt = terminatedAt;
+        _expiringAt = expiringAt;
+    }
+
+    public bool IsMet() => !_terminatedAt.HasValue && _expiringAt > DateTimeOffset.UtcNow;
```
Comment (kamilbaczek) [inline Chapter-4-applying-tactical-domain-driven-design/Fitnet.Contracts/Src/Fitnet.Contracts.Core/AddAnnex/BusinessRules/AnnexCanOnlyBeAddedOnlyBeAddedToActiveBindingContractRule.cs:18]: Using DateTimeOffset.UtcNow here force to use DateTime.Now in unit tests. When the test is executed, result can be non determinictic i would just pass ISystemClock as paramter to bussiness rule and mock it in test with static date. What you think?

**Review by kamilbaczek — CHANGES_REQUESTED** [review]
i've spotted some things that we can improve. Please respond to comments

File: Chapter-4-applying-tactical-domain-driven-design/Fitnet.Contracts/Src/Fitnet.Contracts.Core/AddAnnex/BusinessRules/AnnexCanOnlyBeAddedOnlyBeAddedToActiveBindingContractRule.cs (hunk 2 above)
Comment (meaboutsoftware) [reply Chapter-4-applying-tactical-domain-driven-design/Fitnet.Contracts/Src/Fitnet.Contracts.Core/AddAnnex/BusinessRules/AnnexCanOnlyBeAddedOnlyBeAddedToActiveBindingContractRule.cs]: Done

File: Chapter-4-applying-tactical-domain-driven-design/Fitnet.Contracts/Src/Fitnet.Contracts.Core/AddAnnex/AnnexAddedToBindingContractEvent.cs (hunk 1 above)
Comment (meaboutsoftware) [reply Chapter-4-applying-tactical-domain-driven-design/Fitnet.Contracts/Src/Fitnet.Contracts.Core/AddAnnex/AnnexAddedToBindingContractEvent.cs]: Done

File: Chapter-4-applying-tactical-domain-driven-design/Fitnet.Contracts/Src/Fitnet.Contracts.Core/AddAnnex/BusinessRules/AnnexCanOnlyBeAddedOnlyBeAddedToActiveBindingContractRule.cs (hunk 3 above)
Comment (meaboutsoftware) [reply Chapter-4-applying-tactical-domain-driven-design/Fitnet.Contracts/Src/Fitnet.Contracts.Core/AddAnnex/BusinessRules/AnnexCanOnlyBeAddedOnlyBeAddedToActiveBindingContractRule.cs:18]: But then we should use SystemClock in every single place in our code - we have a lot of different places where we are mixing DateTimeOffset.Now with DateTimeOffset.UtcNow.

Additionally, passing ISystemClock to business rule would require to pass the instance of SystemClock from the entity itself (we manually create the business rule). 

What do you think to handle it outside of this PR in a separate branch and change it everywhere in chapter 4?

Comment (kamilbaczek) [reply Chapter-4-applying-tactical-domain-driven-design/Fitnet.Contracts/Src/Fitnet.Contracts.Core/AddAnnex/BusinessRules/AnnexCanOnlyBeAddedOnlyBeAddedToActiveBindingContractRule.cs:18]: Yeah, the time provider should be already available in our project. Let's use it

**Review by kamilbaczek — APPROVED** [review]

**Pull request merged**